test/metric_test.cc

```diff
@@ -0,0 +1,130 @@
+/*
+http://www.apache.org/licenses/LICENSE-2.0.txt
+Copyright 2016 Intel Corporation
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+#include "snap/config.h"
+#include "snap/metric.h"
+#include "gtest/gtest.h"
+
+#include <chrono>
+#include <ctime>
+#include <sstream>
+#include <string>
+#include <vector>
+
+using std::chrono::system_clock;
+using std::make_pair;
+using std::pair;
+using std::string;
+using Plugin::Metric;
+using Plugin::ConfigPolicy;
+using Plugin::StringRule;
+
+
+static string extract_ns(const Metric& metric) {
+    string ns_str;
+    for (auto const& elem : metric.ns()) {
+        ns_str += "/" + elem.value;
+    }
+    return ns_str;
+}
+
+TEST(MetricTest, SetNsUnitDescriptionWorks) {
+    Metric fake_metric{
+            {
+                    {"foo", "", ""},
+                    {"bar", "", ""},
+            },
+            "atoms",
+            "critical metric"
+    };
+
+    EXPECT_EQ("/foo/bar", extract_ns(fake_metric));
+    EXPECT_EQ("atoms", fake_metric.get_rpc_metric_ptr()->unit());
+    EXPECT_EQ("critical metric", fake_metric.get_rpc_metric_ptr()->description());
```

Comment:
Could you create fake metric with tags and add test for tags (to check `fake_metric.get_rpc_metric_ptr()->tags()`?